xla/tests/collective_ops_test_e2e.cc

```diff
@@ -645,5 +645,81 @@ TEST_F(CollectiveOpsTestE2E, NoAllToAllDecomposition) {
   LiteralTestUtil::ExpectR1Equal<uint32_t>({20, 25, 21, 26}, results[1]);
 }
 
+TEST_F(CollectiveOpsTestE2E, WindowedEinsumE2EAllgatherMultiConsumer) {
```

Comment:
Could you please verify that this test goes through the newly added code path? 

If possible (in your testing infrastructure), you can dump the HLO after each pass and (manually) check the HLO right before and after the new pass. (You need environment variable XLA_FLAGS=--xla_dump_to=<directory> --xla_dump_hlo_pass_re=.* to dump the HLO after each pass.)

Reply:
Thanks @seherellis  for pointing it out. I verified the hlo when i first added this test but I think there was an issue when I was asked to minimize this test, when i removed the gemm+rs from the original hlo, the allgather didnt happen on the shared operand for the second dot anymore, in turn spmd partitioner decided to put ag on the other non-shared operand. I have added back the gemm+rs to the hlo and now it triggers the code path.
I also reduced the size of tensors so this runs a bit faster.
I have also verified the hlo again that the pattern is expected.
It'd be good to keep the gemm+rs pattern in there since we didnt really have any e2e for that before.